internal/backup_push_handler.go

```diff
@@ -70,6 +77,14 @@ type BackupConfig struct {
 	userData                  string
```

Comment:
It is still quite a huge struct, maybe decompose it like this?
```
type CmdArguments struct {
	isPermanent           bool
	verifyPageChecksums   bool
	storeAllCorruptBlocks bool
	tarBallComposerType   TarBallComposerType
	userData              string
	forceIncremental      bool
}

type IncrementSetup struct {
	previousBackupName        string
	previousBackupSentinelDto BackupSentinelDto
	incrementCount            int
}

type CurrentBackupDetails struct {
	backupName               string
	currentBackupSentinelDto *BackupSentinelDto
	currentMeta              ExtendedMetadataDto
	startLSN                 uint64
	endLSN                   uint64
	uncompressedSize         int64
	compressedSize           int64
	systemIdentifier         *uint64
}

type BackupConfig struct {
	uploader        *WalUploader
	pgVersion       int
	pgDataDirectory string
	backupsFolder   string

	currentBackup  CurrentBackupDetails
	incrementSetup IncrementSetup
	arguments      CmdArguments
}
```

Reply:
Yeah, I will change that once I have the unit test working.

Reply:
I started with the structure you proposed but made some extra changes.
With this structure, it makes more sense to have methods instead of functions.
This is the final working set. Please have another look, thanks.